mkosi/config.py

```diff
@@ -44,13 +47,43 @@ def parse_boolean(s: str) -> bool:
     die(f"Invalid boolean literal: {s!r}")
 
 
+def parse_path(value: str, required: bool, absolute: bool = True, expanduser: bool = True, expandvars: bool = True) -> Path:
```

Comment:
Not a fan of so many flags, at least 
```suggestion
def parse_path(value: str, required: bool, *, absolute: bool = True, expanduser: bool = True, expandvars: bool = True) -> Path:
```
or better
```suggestion
def parse_path(value: str, *, required: bool = True/False, absolute: bool = True, expanduser: bool = True, expandvars: bool = True) -> Path:
```

Reply:
No need for default values after `, *,` in which case the parameter must be specified and must be named.

Reply:
Yes, sure, but having one mandatory keyword-only argument and three optional ones makes for a somewhat less uniform API, choosing a default (probably false, maybe all of them should be false by default) makes it more uniform because it handles all the keyword-only arguments on the same footing.

Reply:
The default `True` value for the other parameters makes sense because there are only a few callers using `False`, so they are clearly the exception.

When I first introduced the `expanduser` and `expandvars` parameters, I used `False` as default. The result was a lot of parameters when configuring `search_extra_paths` and an inconsistency between that option and all the others for no obvious reason. Then I switched to `True` and it was clearer that `expandvars` can always be enabled and `expanduser` must be disabled for the targets of `parse_source_target_paths`.

However, I don't feel like there is a clear default for `required`. Some use `True`, some use `False`, and I find it useful that it's explicit when reading `MkosiConfigParser.SETTINGS`.